Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -595,4 +595,301 @@ ROMol *fragmentOnBRICSBonds(const ROMol &mol) {
 }
 } // End of MolFragmenter
-} // end of namespace RDKit
+
+namespace {
+const unsigned int NOLABEL = std::numeric_limits<unsigned int>::max();
+// Get the atom label - this might be useful as a util class
+unsigned int get_label(const Atom *a, const MolzipParams &p) {
+  PRECONDITION(a, "bad atom in MolZip::get_label")
+  unsigned int idx = NOLABEL;
+  switch(p.label) {
+  case MolzipLabel::AtomMapNumber:
+          if(a->getAtomicNum() == 0) {
+              return a->getAtomMapNum();
+          }
+  case MolzipLabel::Isotope:
+          if(a->getAtomicNum() == 0) {
+              return a->getIsotope();
+          }
+  case MolzipLabel::AtomType: {
+    idx = std::distance(p.atomSymbols.begin(),
+                        std::find(p.atomSymbols.begin(), p.atomSymbols.end(),
+                                  a->getSymbol()));
+    if(idx == p.atomSymbols.size()) {
+        idx = NOLABEL;
+    }
+    break;
+  case MolzipLabel::FragmentOnBonds:
+      // shouldn't ever get here
+      assert(0);
+      break;
+  default:
+      CHECK_INVARIANT(0,"bogus MolZipLabel value in MolZip::get_label");
+  }
+  }
+  return idx;
+}
+
+// Return the connected atom
+//  n.b. There can be only one connection from a mapped atom
+Atom *get_other_atom(Atom *a) {
+  PRECONDITION(a, "null atom in MolZip::get_other_atom");
+  auto &m = a->getOwningMol();
+  if(m.getAtomDegree(a) != 1)
+    return nullptr;
+  
+  for(const auto &nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(a))) {
+    return m[nbrIdx];
+  }
+  return nullptr; // can't get here
+}
+
+int num_swaps_to_interconvert(std::vector<unsigned int> &orders) {
+    int nswaps = 0;
+    std::vector<bool> seen(orders.size());
+    for(size_t i=0;i<orders.size();++i) {
+        if(!seen[i]) {
+            auto j = i;
+            while(orders[j] != i) {
+                j=orders[j];
+                seen[j] = true;
+                nswaps++;
+            }
+        }
+    }
+    return nswaps;
+}
+
+// Simple bookkeeping class to bond attachments and handle stereo
+struct ZipBond {
+    Atom *a;       // atom being bonded
+    Atom *a_dummy; // Labelled atom, i.e. [*:1]-C  will bond the C to something
+    Atom *b;       // atom being bonded
+    Atom *b_dummy; // Labelled atom, i.e. [*:1]-O will bond the O to something
+    
+    // Mark the original order of the nbr atoms including the dummy
+    //  The goal is to copy the dummy chiral order over to the
+    //  atom being bonded
+    void mark(Atom *chiral_atom, Atom *dummy_atom, Atom *new_atom) const {
+        if(chiral_atom->getChiralTag()) {
+            std::string mark = "__molzip_mark_" + std::to_string(chiral_atom->getIdx());
+            chiral_atom->setProp("__molzip_chiral_mark", mark);
+            int order = 0;
+            auto &m = chiral_atom->getOwningMol();
+            for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+                m[nbrIdx]->setProp<int>(mark, order);
```

Comment:
It's a nit, but the `<int>` shouldn't be necessary here
```suggestion
                m[nbrIdx]->setProp(mark, order);
```

Reply:
I can change it, but I like being explicit with ints.